MLResurrection.cpp

```diff
@@ -0,0 +1,668 @@
+    wy = dead_cells[index][2]*d+offsety;
+  }
+  else {                      // French side
+    wx = -(dead_cells[index][1]*d+offsetx);
+    wy = dead_cells[index][2]*d+offsety;
+  }
+
+  const geomtools::geom_id& gid = geomtools::geom_id(1204, 0, dead_cells[index][0], dead_cells[index][1], dead_cells[index][2]); // 1204 is the type of a geiger cell core
+
+  if (!calData.calibrated_tracker_hits().empty()) // get the last hit_id from the existing hits
+    id = calData.calibrated_tracker_hits().back()->get_hit_id()+index+1;
+
+  new_calibrated_tracker_hit.set_r(rad);
+  new_calibrated_tracker_hit.set_sigma_r(rad_err);
+  new_calibrated_tracker_hit.set_z(wz);
+  new_calibrated_tracker_hit.set_sigma_z(wz_err);
+  new_calibrated_tracker_hit.set_xy(wx, wy);
+  new_calibrated_tracker_hit.set_delayed(false);
+  new_calibrated_tracker_hit.set_geom_id(gid);
+  new_calibrated_tracker_hit.set_hit_id(id);
+
+  // write hit
+  new_calibrated_data->calibrated_tracker_hits().push_back(new_hit_handle);
+  hits++;
+
+}
+
+void MLResurrectionModule::initialize(const datatools::properties& myConfig,
+			    datatools::service_manager& flServices,
+			    dpp::module_handle_dict_type& /*moduleDict*/){
+
+  // Extract the filename_out key from the supplied config, if the key exists.
+  // datatools::properties throws an exception if the key isn't in the config, so catch this if thrown
+  try {
+    myConfig.fetch("filename_out", filename_output);
+  }
+  catch (std::logic_error& e) {
+    std::cerr << "Problem in the output file " << e.what() << std::endl;
+  };
+
+  try {
+    myConfig.fetch("dead_cells", dead_cells_list);
+  }
+  catch (std::logic_error& e) {
+    std::cerr << "Error in configuration files: file with list of dead cells not present " << e.what() << std::endl;
+  };
+
+  try {
+    myConfig.fetch("method", method);
+  }
+  catch (std::logic_error& e) {
+    std::cerr << "Error in configuration files: method not defined " << e.what() << std::endl;
+  };
+
+  std::string string_filename = std::string("reco_hits_")+method+std::string(".txt");
+  const char *reco_hits_filename = string_filename.c_str();
+  std::rename("reco_hits.txt", reco_hits_filename);
+  reco_hits_file << "ID SIDE LAYER ROW" <<std::endl;
+
+  std::cout << "MLResurrection module initialized..." << std::endl;
+  std::cout << "Method selected: " << method << std::endl;
+  std::cout << "Output file: " << filename_output << std::endl;
+
+  flSimParameters = FLSimulate::FLSimulateArgs::makeDefault();
+
+  datatools::service_manager services("DeadCells", "SuperNEMO Dead Cells");
+  datatools::properties services_config;
+  services_config.store("name", "DeadCells");
+  services_config.store("description","SuperNEMO Demonstrator Dead Cells module");
+  std::vector<std::string> scf = {"@falaise:snemo/demonstrator/geometry/" "GeometryService.conf"};
+  services_config.store("services.configuration_files", scf);
+  services.initialize(services_config);
```

Comment:
What is this for? A service manager is already provided to the `initialise` call, and `services` appears used nowhere else.